examples/connected-components/Makefile

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2024, The Pallene Developers
+# Pallene Tracer is licensed under the MIT license.
+# Please refer to the LICENSE and AUTHORS files for details
+# SPDX-License-Identifier: MIT
+
+OUTPUT_FILE = component.so
+
+CC         = cc
+CFLAGS     = -O2 -std=c99 -pedantic -Wall -Wextra -Wformat-security
+LIBFLAGS   = -fPIC -shared
+
+ifneq ($(findstring debug, $(MAKECMDGOALS)), )
+	CFLAGS += -DPT_DEBUG
+endif
```

Comment:
if statements in makefiles are complicated and non-portable. Consider

* Just turn on DPT_DEBUG all the time
* Configure the DPT_DEBUG via variables

Another alternative is to recursivelly invoke make, just like Lua's makefile does for `make linux`. But IMO that might be too complicated for what we are doing here.